index.html

```diff
@@ -149,11 +149,15 @@ <h3>
         </h3>
         <p>
           The [=note_taking=] <code>new_note_url</code> member is a <a>URL</a>
-          [=manifest/within scope=] of the manifest, to open when the user
-          wants to take a new note using the web application (e.g., from an
-          operating system shortcut icon or keyboard shortcut). Defining this
-          member indicates that the web application is a note-taking
-          application.
+          to open when the user wants to take a new note using the web
+          application (e.g., from an operating system shortcut icon or keyboard
+          shortcut). Defining this member indicates that the web application is
+          a note-taking application.
+        </p>
+        <p>
+          The <code>new_note_url</code> member is parsed with
```

Comment:
This isn't formal enough (e.g., doesn't actually say how to invoke the parser, doesn't say what to do iff it's not within scope).

"[process a shortcut](https://www.w3.org/TR/appmanifest/#dfn-process-a-shortcut)" is a good example of how to specify processing of a must-be-within-scope URL in the Manifest.

Quick note about the word "must": this is a special word (along with "required", "shall", "should", "recommended", "may" and "optional") defined in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119) which has a specific meaning in normative language -- it imposes a requirement on the user agent which implements this spec. (So if we use it, it should always be capitalized, and should not appear in non-normative text.) The problem in this case is that you aren't imposing a requirement on the user agent, you're imposing a requirement on the manifest. We don't impose requirements on the documents that our specs process; rather, we say exactly what the user agent should do in the event that the document isn't valid. (So instead of saying "must be within scope", we give instructions for what to do if you find a URL that isn't within scope.)

So what should be done here:

1. I think put a separate section at the top of this document (above `display_override`) called "Extension to processing the manifest", which says "The following steps are added to the extension point in the steps for processing a manifest:" and then says "Process the `new_note` member, passing _json_ and _manifest_."
2. Here, create an algorithm called "To process the `new_note` member, given object _json_ and ordered map _manifest_:". In the body, invoke the URL parser on _json_`["new_note"]`; if it's failure, return (silently dropping the `new_note` member); otherwise if it isn't within scope, return (silently dropping); otherwise store the result in _manifest_`["new_note"]`. (Use "[process a shortcut](https://www.w3.org/TR/appmanifest/#dfn-process-a-shortcut)" as a guide.)

It turns out that (news to me) now every member needs an explicit processing step, not just ones with special requirements. Which means we actually need to add such a step for all of the other ones in peripheral specs, like `display_override`, `share_target`, etc. But let's leave that out for this PR.

Reply:
I've moved the non-normative descriptive bits to a "Note" section and used MAY for the "treat the web application differently" and how to treat the new_note_url (similar to how start_url is defined).

I've added explicit processing steps including adding this as an extension to the manifest processing extension point. I've put it inline in the note_taking processing section instead of at the top of the doc so that each extension is independent of others.